source/extensions/watchdog/profile_action/profile_action.cc

```diff
@@ -54,6 +67,7 @@ void ProfileAction::run(
   if (running_profile_) {
     return;
   }
+  profiles_attempted_.inc();
```

Comment:
Consider moving down to line 95 or 96:

if (!Profiler::Cpu::profilerEnabled()) {
  profiles_attempted_.inc();
  if (Profiler::Cpu::startProfiler(profile_filename_)) {

or

if (!Profiler::Cpu::profilerEnabled()) {
  if (Profiler::Cpu::startProfiler(profile_filename_)) {
    profiles_attempted_.inc();

Reply:
I originally thought doing what you suggested, but my reason for putting `profiles_attempted_.inc()` where it is is that it gives us a broader view of potential errors that might be occurring. 

It would provide more information when monitoring these two counters, and if they diverge too much could be a variety of causes such as we're reached the limit for profiles, someone accidentally deleted the directory, the file didn't save to disk, etc.

If it was in the places suggested, it would likely have less divergence, but pinpoint particular issues.
For the 1st suggestion it would track if there's contention on the profiler or if the profile failed after starting
For the 2nd suggestion it would track if the profile failed after starting

So my question then is, do we want it to be more broad or do we want it to pinpoint issues?

Reply:
Like you say, it depends on what stats we want to provide and what are their semantics.  The most important stat added by this PR is the number of profiles successfully collected.

I'm fine with the attempts being incremented in the 3 places described above, or simply not having the stat.

Reply:
Sgtm. Will leave it as-is for now since I think the additional errors it can catch could be useful for monitoring the mechanism.